third_party/WebIDL/WebIDL.py

```diff
@@ -43,18 +44,28 @@ def parseInt(literal):
     return value * sign
 
 
+class CustomEnum(Protocol):
+    attrs: tuple[str, ...]
+
+    def __getattr__(self, name: str) -> int: ...
+
 # This is surprisingly faster than using the enum.IntEnum type (which doesn't
 # support 'base' anyway)
-def enum(*names, base=None):
+def enum(*names: str, base = None) -> CustomEnum:
```

Comment:
I don't think it is a good idea to make such large changes to the definition of this method. e.g the comment does say that the `CustomEnumType` is a performance optimization and we are replacing it with `_getattr_` implementation that is based on `list.index`, which will we be much slower than simple field access.

I think we should just go with annotation `Tags` as `Any` type in the `IDLType` class.